docker/scripts/p4runtime_lib/switch.py

```diff
@@ -0,0 +1,191 @@
+
+MSG_LOG_MAX_LEN = 1024
+
+# List of all active connections
+connections = []
+
+def ShutdownAllSwitchConnections():
+    for c in connections:
+        c.shutdown()
+
+class SwitchConnection(object):
+
+    def __init__(self, name=None, address='127.0.0.1:50051', device_id=0,
+                 proto_dump_file=None):
+        self.name = name
+        self.address = address
+        self.device_id = device_id
+        self.p4info = None
+        self.channel = grpc.insecure_channel(self.address)
+        if proto_dump_file is not None:
+            interceptor = GrpcRequestLogger(proto_dump_file)
+            self.channel = grpc.intercept_channel(self.channel, interceptor)
+        self.client_stub = p4runtime_pb2.P4RuntimeStub(self.channel)
+        self.requests_stream = IterableQueue()
+        self.stream_msg_resp = self.client_stub.StreamChannel(iter(self.requests_stream))
+        self.proto_dump_file = proto_dump_file
+        connections.append(self)
+
+    @abstractmethod
+    def buildDeviceConfig(self, **kwargs):
+        return p4config_pb2.P4DeviceConfig()
+
+    def shutdown(self):
+        self.requests_stream.close()
+        self.stream_msg_resp.cancel()
+
+    def MasterArbitrationUpdate(self, dry_run=False, **kwargs):
+        request = p4runtime_pb2.StreamMessageRequest()
+        request.arbitration.device_id = self.device_id
+        request.arbitration.election_id.high = 0
+        request.arbitration.election_id.low = 1
+
+        if dry_run:
+            print "P4Runtime MasterArbitrationUpdate: ", request
+        else:
+            self.requests_stream.put(request)
+            for item in self.stream_msg_resp:
+                return item # just one
+
+    def SetForwardingPipelineConfig(self, p4info, dry_run=False, **kwargs):
+        device_config = self.buildDeviceConfig(**kwargs)
+        request = p4runtime_pb2.SetForwardingPipelineConfigRequest()
+        request.election_id.low = 1
+        request.device_id = self.device_id
+        config = request.config
+
+        config.p4info.CopyFrom(p4info)
+        config.p4_device_config = device_config.SerializeToString()
+
+        request.action = p4runtime_pb2.SetForwardingPipelineConfigRequest.VERIFY_AND_COMMIT
+        if dry_run:
+            print "P4Runtime SetForwardingPipelineConfig:", request
+        else:
+            self.client_stub.SetForwardingPipelineConfig(request)
+
+    def WriteTableEntry(self, table_entry, dry_run=False):
+        request = p4runtime_pb2.WriteRequest()
+        request.device_id = self.device_id
+        request.election_id.low = 1
+        update = request.updates.add()
+        if table_entry.is_default_action:
+            update.type = p4runtime_pb2.Update.MODIFY
+        else:
+            update.type = p4runtime_pb2.Update.INSERT
+        update.entity.table_entry.CopyFrom(table_entry)
+        if dry_run:
+            print "P4Runtime Write:", request
+        else:
+            self.client_stub.Write(request)
```

Comment:
I believe this class should have a `Write` method which would call `self.client_stub.Write`. That class could take care of populating the `device_id` and `election_id` fields. **It could also be used by an app to directly send P4Runtime `WriteRequest` messages, which could be useful if an app wants to use P4Runtime directly or use P4Runtime for things that are not covered by this helper library.**